mozilla_django_oidc/views.py

```diff
@@ -114,15 +114,15 @@ def redirect_url(self):
 
     def dispatch(self, request, *args, **kwargs):
         """Log out the user."""
-
-        if request.user.is_authenticated():
+        # Check if a method exists to build the URL to log out the user
+        # from the OP.
+        logout_from_op = import_from_settings('OIDC_OP_LOGOUT_URL_METHOD', '')
```

Comment:
> I agree--if the user isn't logged in to the app, then this view should essentially be a no-op.

But it's not that simple. If the user isn't logged in, sure, you don't call `auth.logut()` but you still need to redirect somewhere. If `OIDC_OP_LOGOUT_URL_METHOD` assumes that the user is authenticated, then it'd be weird to bypass that and fallback on the default (`self.redirect_url`). 

Options:

1. Change the *documentation* such that... "If you implement and set 'OIDC_OP_LOGOUT_URL_METHOD' you can not assume that the user is logged in."
2. My other suggestion above about exiting early with a `HttpResponseBadRequest`. 
3. Return a `403 Forbidden` if someone tries to log out when they're not logged out. 
4. Refactor the patch so that `OIDC_OP_LOGOUT_URL_METHOD` is only invoked when the user is logged in.